### DISCUSSION TITLE
Hide some user's fields to non administrator?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#10907] [Archive]
Is there a way to hide some fields and to restrict the list to only the users with a certain role?
If it is not possible, another way be rely on a view from the database?

**Comment by azrikahar** [comment]
> Is there a way to hide some fields

Just to clarify, do you mean fields in the user page? You can modify the read permission of the role and limit the fields via Field Permissions, but do note that there's a set of fields that are required for users with app access as seen here:

![image](https://user-images.githubusercontent.com/42867097/148556433-2a41d243-4975-4922-ad79-2891969a6847.png)

> and to restrict the list to only the users with a certain role?

You can pair Item Permissions with [dynamic variables such as $CURRENT_ROLE](https://docs.directus.io/configuration/filter-rules/#dynamic-variables) to do that 👍

**Reply by Pachat** [reply]
Great! Thanks!

BTW, 
- looks like `last_page` belongs to the minimum permissions (wonder why?) though not in the list, and
- to follow the shown order of the fields to check, `avatar` is before `location`

**Reply by azrikahar** [reply]
> looks like `last_page` belongs to the minimum permissions (wonder why?) though not in the list

Seems like it's present on API side but is not reflected on the list shown on App side. Added a PR for it. Thanks!

> to follow the shown order of the fields to check, `avatar` is before `location`

I believe this is just the original ordering of the fields before it was changed. Technically won't impact anything, but ya could be updated if it ended up being too confusing.
